backend/googlecloudstorage/googlecloudstorage.go

```diff
@@ -743,7 +756,41 @@ func (f *Fs) PutStream(ctx context.Context, in io.Reader, src fs.ObjectInfo, opt
 // Mkdir creates the bucket if it doesn't exist
 func (f *Fs) Mkdir(ctx context.Context, dir string) (err error) {
 	bucket, _ := f.split(dir)
-	return f.makeBucket(ctx, bucket)
+	e := f.makeBucket(ctx, bucket)
+	if e != nil {
+		return e
+	}
+	if f.opt.DirectoryMarkers {
+		return f.uploadDirFile(ctx, bucket, dir)
+	}
+	return nil
+}
+
+// Creates empty file as directory marker
+func (f *Fs) uploadDirFile(ctx context.Context, bucket string, dir string) (err error) {
```

Comment:
Can you call Object.Update here instead of duplicating the upload code?

Reply:
Also done, required o.split change